### ISSUE TITLE
Expose some basic Scholia traffic stats

### ISSUE THREAD

**Issue by Daniel-Mietchen** [WDscholia/scholia#458]
to show usage and as a criterion to help prioritize further development.
For instance, now that Commons and some Wikipedias link to Scholia, do we see people actually use it?

**Comment by fnielsen** [comment]
Yes, that would be nice. 

We need to think of privacy:
* A user accidentially has sensitive information in the URL. We counter this problem by only recording URLs that fit a correct pattern.
* Given that the usage of Scholia is not yet so large, it might be that on certain times one is able to track the user, particular if the second precision is maintained. Perhaps hourly or daily aggregation is sufficient for a user to "hide in the entropy".

**Comment by Daniel-Mietchen** [comment]
For a while, the only thing we've had in this regard was [this figure](https://doi.org/10.3897/rio.5.e35820.figure3) but now at WMhack, there seems to be progress:
- [daily traffic stats for (all?) Toolforge tools](https://tools.wmflabs.org/toolviews/api/v1/day/2019-04-23) ([source tweet](https://twitter.com/fnielsen/status/1129702729960570880))
- [demo visualization](https://twitter.com/fnielsen/status/1129730557951324160) of these stats for Scholia, Wembedder and Ordia

**Comment by fnielsen** [comment]
People are working on a general tool for Toolforge tool views. 

At the Wikimedia Hackathon 2019 a WMF employee told about an API for daily views. 

https://tools.wmflabs.org/toolviews/api/v1/day/2019-04-23

This is for all tools. We would like to have the data transposed to we can view all stats for a single tool. I is my impression that there is some contact to the pageview tools developer. Perhaps they will do something with the general data. 

Of course we may be interested in more detailed Scholia data, e.g., aspect statistics.

**Comment by fnielsen** [comment]
Credientials for database for toolviews is here: https://phabricator.wikimedia.org/source/tool-toolviews/browse/master/toolviews/__init__.py

```bash
$ mysql -u <username> s53734__toolviews_p -p
```

```SQL
select * from daily_raw_views where tool = 'scholia';
```

There are statistics from 2018-04-29.

**Comment by fnielsen** [comment]
https://twitter.com/fnielsen/status/1145984623563935744

**Comment by carlinmack** [comment]
Here's a graph of Scholia usage, you can change the date via the URL at the top https://codepen.io/carlinmack/pen/LYyBdZo 

D3 is tricky to use without a plotting framework built on top of it (like Observable Plot (ISC) or Plotly JS (MIT))

**Comment by fnielsen** [comment]
Does plotly.js rely on servers on plotly.com? (I may have misunderstood this)

**Comment by carlinmack** [comment]
it just relies on plotly.min.js, which I've loaded from their CDN for convenience

**Comment by fnielsen** [comment]
Yes, that was also what I could see on codepen. The license is MIT License. Maybe we should consider that. We have also been mentioning raw D3 and nvd3. In Plotly, it seems also to be possible to do links (https://stackoverflow.com/questions/25148462/open-a-url-by-clicking-a-data-point-in-plotly)

**Comment by carlinmack** [comment]
I'm happy to mock-up some charts in plotly, nvd3 and observable plot (which is made by the creator of D3) for comparison 👍

**Comment by carlinmack** [comment]
I've looked into the options mentioned above, in #760 and Chart.js and all the examples can be seen in https://codepen.io/carlinmack/pen/LYyBdZo . I've written a comment besides each, I think the best options are Plotly and Apache ECharts, happy to go with either. Sadly both nvd3 and dimple are not maintained :(

**Comment by carlinmack** [comment]
I've now added click events to 3 of the charts - Apache echarts, chart.js and Plotly js - to show that we can link from the charts. Would appreciate thoughts on which to use from anyone :)

**Comment by carlinmack** [comment]
I think we should discuss in #760 rather than this issue
